src/main/java/com/atlassian/db/replica/api/aurora/AuroraReplicasDiscoverer.java

```diff
@@ -0,0 +1,55 @@
+package com.atlassian.db.replica.api.aurora;
+
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static com.atlassian.db.replica.api.aurora.AuroraEndpoints.replicaUrl;
+
+/**
+ * It allows to pull Aurora Replicas cluster information
+ */
+public class AuroraReplicasDiscoverer {
+    private final Connection connection;
+
+    public AuroraReplicasDiscoverer(Connection connection) {
+        this.connection = connection;
+    }
+
+    /**
+     * It provides list of replica urls based on reader endpoint and server ids
+     * @param readerEndpoint - Aurora reader endpoint
+     * @return list of replicas jdbcUrls
+     * @throws SQLException
+     */
+    public List<String> fetchReplicasUrls(String readerEndpoint) throws SQLException {
+        return fetchReplicaServerIds()
+            .stream()
+            .map(serverId -> replicaUrl(
+                readerEndpoint,
+                serverId
+            )).collect(Collectors.toList());
+    }
+
+    /**
+     * Asks Aurora about the list of server ids
+     * @return server ids
+     * @throws SQLException
+     */
+    public List<String> fetchReplicaServerIds() throws SQLException {
+        LinkedList<String> ids = new LinkedList<>();
+        try (ResultSet rs =
+                 connection.prepareStatement(
+                     "SELECT server_id FROM aurora_global_db_instance_status() WHERE session_id != 'MASTER_SESSION_ID'")
```

Comment:
Is it possible to integration test it in Docker?

Reply:
We don't have Aurora, we could create a function in postgres that will return the same rows all the time. Not ideal, but better than zero coverage, right?

Reply:
Nice thinking outside the box! The function could read from a synthetic table. Mutating the table would be like re-mocking it for different test cases.